Sources/SKTestSupport/BuildServerTestProject.swift

```diff
@@ -59,7 +59,7 @@ package class BuildServerTestProject: MultiFileTestProject {
     testName: String = #function
   ) async throws {
     var files = files
-    files["buildServer.json"] = """
+    files[".bsp/buildServer.json"] = """
```

Comment:
Could you also keep a test around that validates that we keep finding the build server config if it’s placed at `<workspace root>/buildServer.json`. We need to keep supporting that for compatibility reasons.

Reply:
Did you mean to keep the old logic:

```swift
localFileSystem.isFile(workspaceFolder.appending(component: "buildServer.json"))
```

Reply:
Ye

Reply:
Currently we have multiple `.json` files in the following paths:

- `<workspace root>/buildServer.json`
- `<workspace root>/.bsp/buildServer.json`

In the `.bsp` directory I have included the following too:

```json
".bsp/test.json": "{}"
```

So, which is the preferred location the path as per the old logic or the new one? 

Since, now the test currently fails due to the presence of `test.json` in `.bsp` which is empty and even though we have one in `<workspace root>` it's not used as `getConfigPath` returns early.

Reply:
> So, which is the preferred location the path as per the old logic or the new one?

We should prefer files in `<workspace root>/.bsp`, then look at `<workspace root>/buildServer.json` and then continue looking at the system locations

> As now the test currently fails due to the presence of test.json in .bsp which isn't a valid config file and even though we have one in <workspace root> it's not used as getConfigPath returns early.

I would say this is expected. In case you have a malformed build server config file, we don’t silently fall to system files, which would be surprising, I think.

Reply:
Ok, let me know if any further changes are needed.